plugins/packages/databricks/lib/index.ts

```diff
@@ -0,0 +1,71 @@
+import { ConnectionTestResult, QueryService, QueryResult, QueryError } from '@tooljet-plugins/common';
+import { SourceOptions, QueryOptions } from './types';
+import { DBSQLClient } from '@databricks/sql';
+import IDBSQLSession from '@databricks/sql/dist/contracts/IDBSQLSession';
+import IOperation from '@databricks/sql/dist/contracts/IOperation';
+import Int64 from 'node-int64';
+
+export default class Databricks implements QueryService {
+  async testConnection(sourceOptions: SourceOptions): Promise<ConnectionTestResult> {
+    const client = await this.getConnection(sourceOptions);
+    const session: IDBSQLSession = await client.openSession();
+    const queryOperation: IOperation = await session.executeStatement('SELECT 1', {
+      runAsync: true,
+      queryTimeout: new Int64(10000),
+    });
+    let result;
+    try {
+      result = await queryOperation.fetchAll();
+    } catch (error) {
+      throw new Error('Error in connection: ' + error.message);
+    } finally {
+      await session.close();
+      await client.close();
+    }
+
+    return {
+      status: 'ok',
+      data: result,
+    };
+  }
+  async getConnection(sourceOptions: SourceOptions): Promise<DBSQLClient> {
+    const credentials = {
+      host: sourceOptions.host,
+      path: sourceOptions.http_path,
+      token: sourceOptions.personal_access_token,
+      socketTimeout: 3,
+    };
+    try {
+      const client = new DBSQLClient();
+      client.connect(credentials);
+      client.on('error', (error) => {
+        throw new Error('Error in connection: ' + error.message);
```

Comment:
Please simply log the error as throwing error explicitly from here won't be caught
`console.error`